crates/weaver_semconv/src/v2/mod.rs

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! The new way we want to define data going forward.
+
+use std::collections::BTreeMap;
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    deprecated::Deprecated,
+    group::GroupSpec,
+    semconv::{Imports, SemConvSpecV1},
+    stability::Stability,
+    v2::{attribute::AttributeDef, entity::Entity, event::Event, metric::Metric, span::Span},
+    YamlValue,
+};
+
+pub mod attribute;
+pub mod entity;
+pub mod event;
+pub mod metric;
+pub mod span;
+
+/// Common fields we want on all major components of semantic conventions.
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct CommonFields {
+    /// A brief description of the span.
+    pub brief: String,
+    /// A more elaborate description of the span.
+    /// It defaults to an empty string.
+    #[serde(default, skip_serializing_if = "String::is_empty")]
+    pub note: String,
+    /// Specifies the stability of the span.
+    pub stability: Stability,
+    /// Specifies if the semantic convention is deprecated. The string
+    /// provided as description MUST specify why it's deprecated and/or what
+    /// to use instead. See also stability.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    pub deprecated: Option<Deprecated>,
+    /// Annotations for the group.
+    #[serde(default, skip_serializing_if = "BTreeMap::is_empty")]
+    pub annotations: BTreeMap<String, YamlValue>,
```

Comment:
Elsewhere annotations are defined as Option. We should be consistent.

Reply:
They're define as Option because they can be an override.  For the purpose of provenance, should we just assume an empty tree is the same as not defining anything?  Should override always be *additive* as opposed to replacing the previous value?